frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,163 @@
+  filters: { [key: string]: any };
+  listSorts: any;
+};
+
+type props = MonitoringAlertsProps & StateProps;
+
+const POLL_DELAY = 15 * 1000;
+const reduxID = 'devMonitoringAlerts';
+const textFilter = 'resource-list-text';
+
+export const MonitoringAlerts: React.FC<props> = ({ match, rules, filters, listSorts }) => {
+  const [sortBy, setSortBy] = React.useState({});
+  const [rows, setRows] = React.useState([]);
+  const [collapsedRows, setCollapsedRows] = React.useState([]);
+  const dispatch = useDispatch();
+  const namespace = match.params.ns;
+  const { sortBy: listSortBy, orderBy: listOrderBy } = getURLSearchParams();
+  const columnIndex = _.findIndex(monitoringAlertColumn, { title: listSortBy });
+  const sortOrder = listOrderBy || SortByDirection.asc;
+  const [response, loadError, loading] = useURLPoll<PrometheusRulesResponse>(
+    `${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/rules?namespace=${namespace}`,
+    POLL_DELAY,
+    namespace,
+  );
+  const thanosRules = React.useMemo(
+    () => (!loading && !loadError ? getAlertsAndRules(response?.data).rules : []),
+    [response, loadError, loading],
+  );
+
+  React.useEffect(() => {
+    const sortThanosRules = _.sortBy(thanosRules, (rule) => alertingRuleStateOrder(rule));
+    dispatch(monitoringSetRules('devRules', sortThanosRules));
+  }, [dispatch, thanosRules]);
+
+  const filteredRules = React.useMemo(() => {
+    const filtersObj = filters?.toJS();
+    const listSortsObj = listSorts?.toJS() || {};
+    if (columnIndex > -1) {
+      setSortBy({ index: columnIndex + 1, direction: sortOrder });
+    }
+    const fRules = getFilteredRows(filtersObj, alertFilters, rules);
+    const { field, orderBy } = listSortsObj;
+    return applyListSort(fRules, orderBy || listOrderBy, field || _.camelCase(listSortBy));
+  }, [listSorts, columnIndex, sortOrder, filters, rules, listOrderBy, listSortBy]);
+
+  React.useEffect(() => {
+    const tableRows = monitoringAlertRows(filteredRules);
+    _.forEach(tableRows, (tr) => {
+      _.forEach(collapsedRows, (cr) => {
+        if (tr.cells[0].id === cr.cells[0].id) {
+          tr.isOpen = cr.isOpen;
+        }
+      });
+    });
+    setRows(tableRows);
+  }, [collapsedRows, filteredRules]);
+
+  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
+    rows[rowKey].isOpen = isOpen;
+    const collapsedRowsIndex = _.findIndex(
+      collapsedRows,
+      (row) => row.cells[0].id === rows[rowKey].cells[0].id,
+    );
+    if (collapsedRowsIndex === -1) {
+      setCollapsedRows([...collapsedRows, rows[rowKey]]);
+    } else if (collapsedRowsIndex > -1 && isOpen) {
+      collapsedRows.splice(collapsedRowsIndex, 1);
+      setCollapsedRows(collapsedRows);
+    }
+    setRows([...rows]);
+  };
+  const handleSort = (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
+    switch (index) {
+      case 1:
+        dispatch(sortList(reduxID, 'name', undefined, direction, 'Name'));
```

Comment:
I don't think we should be sorting the data in redux. Now we are sorting both in redux with `dispatch(sortList(...))` and we are sorting here in the render function with `applyListSort`.

Also, the sorting only applies to this list, so it doesn't need to be in redux.

A simpler approach would be to just sort directly when rendering with something like
```
const sortedRules = React.useMemo(() => {
  const sorted = _.sortBy(filteredRules, (rule) => {
    switch (sortBy.index) {
      ...
    }
  });
  return sortBy.direction === SortByDirection.desc ? sorted.reverse() : sorted;
}, [filteredRules, sortBy]);
```

Then the handler would be just
```
const handleSort = (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
  setSortBy({ index, direction });
};
```